sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuAggregateExec.scala

```diff
@@ -1644,19 +1654,105 @@ object GpuTypedImperativeSupportedAggregateExecMeta {
           // GpuRowToColumnarExec
           case List(parent, child) if parent.canThisBeReplaced =>
             val childPlan = child.wrapped.asInstanceOf[SparkPlan]
-            val expressions = createBufferConverter(stages(i), stages(i + 1), true)
-            childPlan.setTagValue(GpuOverrides.preRowToColProjection, expressions)
+            val expressions = createBufferConverter(stages(i), stages(i + 1),
+              fromCpuToGpu = true)
+            // Insert the converter into childPlan as a tag value
+            insertBufferConverter(childPlan, expressions, isR2C = true)
```

Comment:
NIT: It would be nice to unify the argument name, i mean `isR2C` -> `fromCpuToGpu`.